src/SamUser/Entity/User.php

```diff
@@ -230,6 +231,44 @@ public function getRoles()
      */
     public function addRole($role)
     {
-        $this->roles[] = $role;
+        $this->roles->add($role);
+    }
+    
+    /**
+     * Remove a role from the user.
+     *
+     * @param Role $role
+     *
+     * @return void
+     */
+    public function removeRole($role)
+    {
+        $this->roles->removeElement($role);
+    }
+    
+    /**
+     * Add roles to the user.
+     *
+     * @param Collection $roles
+     *
+     * @return void
+     */
+    public function addRoles(Collection $roles)
+    {
+        foreach ($roles as $role)
+            $this->roles->add($role);
```

Comment:
Even though it's valid PHP wise, this doesn't go hand in hand with PSR-2. Please use braces even if the body is just one line ;) https://github.com/php-fig/fig-standards/blob/master/accepted/PSR-2-coding-style-guide.md#55-foreach